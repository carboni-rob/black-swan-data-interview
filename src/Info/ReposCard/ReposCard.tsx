import React from "react";

type Props = {
  repos: any[];
};

const ReposCard = ({ repos }: Props): JSX.Element => {
  return (
    <div className="card">
      <h2>User Repositories: {repos.length}</h2>
      <div className="cardContent">
        {repos.map(
          (repo: any): JSX.Element => {
            return (
              <a href={repo.clone_url} key={repo.id}>
                {repo.name}
              </a>
            );
          }
        )}
      </div>
    </div>
  );
};

export default ReposCard;
