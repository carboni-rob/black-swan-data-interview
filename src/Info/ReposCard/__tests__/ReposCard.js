/* eslint-disable @typescript-eslint/camelcase */
import React from "react";
import { render } from "@testing-library/react";
import ReposCard from "../ReposCard";

describe("<ReposCard />", () => {
  const repos = [
    {
      id: 1,
      name: "test1",
      clone_url: "test1/repos/url"
    },
    {
      id: 2,
      name: "test2",
      clone_url: "test2/repos/url"
    },
    {
      id: 3,
      name: "test3",
      clone_url: "test3/repos/url"
    }
  ];

  const tree = <ReposCard repos={repos} />;
  it("should match last snapshot", () => {
    const { container } = render(tree);

    expect(container.firstChild).toMatchSnapshot();
  });
});
