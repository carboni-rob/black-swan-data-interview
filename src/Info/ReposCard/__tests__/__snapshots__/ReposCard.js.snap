// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposCard /> should match last snapshot 1`] = `
<div
  class="card"
>
  <h2>
    User Repositories: 
    3
  </h2>
  <div
    class="cardContent"
  >
    <a
      href="test1/repos/url"
    >
      test1
    </a>
    <a
      href="test2/repos/url"
    >
      test2
    </a>
    <a
      href="test3/repos/url"
    >
      test3
    </a>
  </div>
</div>
`;
